src/vector.js

```diff
@@ -26,26 +27,147 @@ class Vector {
       this.x = x;
       this.y = y;
     }
+    // @endif
   }
 
   /**
-   * Return a new Vector whose value is the addition of the current Vector and the passed in Vector. If `dt` is provided, the result is multiplied by the value.
+   * Calculate the addition of the current vector with the passed in vector.
    * @memberof Vector
    * @function add
    *
    * @param {Vector|{x: number, y: number}} vector - Vector to add to the current Vector.
-   * @param {Number} [dt=1] - Time since last update.
    *
    * @returns {Vector} A new Vector instance.
    */
-  add(vec, dt = 1) {
+  add(vec) {
     return new Vector(
-      this.x + (vec.x || 0) * dt,
-      this.y + (vec.y || 0) * dt,
+      this.x + vec.x,
+      this.y + vec.y,
       this
     );
   }
 
+  // @ifdef VECTOR_SUBTRACT
+  /**
+   * Calculate the subtraction of the current vector with the passed in vector.
+   * @memberof Vector
+   * @function subtract
+   *
+   * @param {Vector|{x: number, y: number}} vector - Vector to subtract from the current Vector.
+   *
+   * @returns {Vector} A new Vector instance.
+   */
+   subtract(vec) {
+    return new Vector(
+      this.x - vec.x,
+      this.y - vec.y,
+      this
+    );
+  }
+  // @endif
+
+  // @ifdef VECTOR_SCALE
+  /**
+   * Calculate the multiple of the current vector by a value.
+   * @memberof Vector
+   * @function scale
+   *
+   * @param {Number} value - Value to scale the current Vector.
+   *
+   * @returns {Vector} a new Vector instance.
+   */
+  scale(value) {
+    return new Vector(
+      this.x * value,
+      this.y * value
+    );
+  }
+  // @endif
+
+  // @ifdef VECTOR_NORMALIZE
+  // @ifdef VECTOR_LENGTH
+  /**
+   * Calculate the normalized value of the current vector. Requires the Vector (length)[/api/vector#length] function.
+   * @memberof Vector
+   * @function normalize
+   *
+   * @returns {Vector} a new Vector instance.
+   */
+  // @see https://github.com/jed/140bytes/wiki/Byte-saving-techniques#use-placeholder-arguments-instead-of-var
+  normalize(length = this.length()) {
+    return new Vector(
+      this.x / length,
+      this.y / length
+    );
+  }
+  // @endif
+  // @endif
+
+  // @ifdef VECTOR_DOT
+  /**
+   * Calculate the dot product of the current vector with the passed in vector.
+   * @memberof Vector
+   * @function dot
+   *
+   * @param {Vector|{x: number, y: number}} vector - Vector to dot product against.
+   *
+   * @returns {Number} The dot product of the vectors.
+   */
+  dot(vec) {
+    return this.x * vec.x + this.y * vec.y;
+  }
+  // @endif
+
+  // @ifdef VECOTR_LENGTH
+  // TODO: function name due to livingcss error tags:107 (overriding length property of an array causes an error)
+  /**
+   * Calculate the length (magnitude) of the Vector.
+   * @memberof Vector
+   * @function lengthh
+   *
+   * @returns {Number} The length of the vector.
+   */
+  length() {
+    return (this.x * this.x + this.y * this.y) ** 0.5;
```

Comment:
Wouldn't `Math.hypot()` safe some bytes before and after optimization? Does the compiler recognize the hypotenuse function?

```
(this.x * this.x + this.y * this.y)
Math.hypot(this.x, this.y)
```

```
Math.sqrt(this.x * this.x + this.y * this.y)
Math.sqrt(Math.hypot(this.x, this.y))
Math.hypot(this.x, this.y) ** .5
```

Reply:
You know what, I never knew there was a `Math.hypot` function lol.